src/libs/CardUtils.ts

```diff
@@ -251,17 +251,15 @@ function isCustomFeed(feed: CompanyCardFeed): boolean {
     return [CONST.COMPANY_CARD.FEED_BANK_NAME.MASTER_CARD, CONST.COMPANY_CARD.FEED_BANK_NAME.VISA, CONST.COMPANY_CARD.FEED_BANK_NAME.AMEX].some((value) => value === feed);
 }
 
-function getCompanyFeeds(cardFeeds: OnyxEntry<CardFeeds>): CompanyFeeds {
-    return {...cardFeeds?.settings?.companyCards, ...cardFeeds?.settings?.oAuthAccountDetails};
-}
-
-function removeExpensifyCardFromCompanyCards(cardFeeds: OnyxEntry<CardFeeds>): CompanyFeeds {
-    if (!cardFeeds) {
-        return {};
-    }
-
-    const companyCards = getCompanyFeeds(cardFeeds);
-    return Object.fromEntries(Object.entries(companyCards).filter(([key]) => key !== CONST.EXPENSIFY_CARD.BANK));
+function getCompanyFeeds(cardFeeds: OnyxEntry<CardFeeds>, shouldFilterOutRemovedFeeds = false): CompanyFeeds {
+    return Object.fromEntries(
+        Object.entries(cardFeeds?.settings?.companyCards ?? {}).filter(([key, value]) => {
```

Comment:
@VickyStash Why do we remove cardFeeds?.settings?.oAuthAccountDetails here? As I understand, direct feeds will be saved in oAuthAccountDetails

Reply:
Right now, both custom and direct feed settings are stored under the `companyCards` field. `oAuthAccountDetails` stores only data for direct feed card assignment.
See the data:
```
[
    {
        "key": "sharedNVP_private_domain_member_18461622",
        "onyxMethod": "set",
        "value": {
            "settings": {
                "companyCardNicknames": {
                    "oauth.americanexpressfdx.com 1001": "American Express cards1",
                    "vcf": "Visa cards1"
                },
                "companyCards": {
                    "cdf": {
                        "pending": true
                    },
                    "oauth.americanexpressfdx.com 1001": {
                        "asrEnabled": false,
                        "forceReimbursable": "force_no",
                        "liabilityType": "personal",
                        "preferredPolicy": "D6A2FBEF5E601C15",
                        "reportTitleFormat": "",
                        "shouldApplyCashbackToBill": true,
                        "statementPeriodEndDay": 1,
                        "uploadLayoutSettings": []
                    }
                },
                "oAuthAccountDetails": {
                    "oauth.americanexpressfdx.com 1001": {
                        "accountList": [
                            "Business Green Rewards Card - 1001"
                        ],
                        "credentials": "Fv6DIZqi836JYuSLyfsvzwiIQyQSIcfwBu7Mb50pxJsDjEdP2vKee64s6fjokcVu3JYgsQEL2K7qb3ItXY+kOgzyy/syGrQoMHikkvBAqhNUnzfHX1CdFoRXxXtaISS8PliI/tMXy6rnXe8CDYrbHh74lDqLHq9Xy9x4xyZ0tdUfKxC/T8yj1oVzJDQkFzVsQ5oM8cuZWdTPni9EoSKlF7q8xgwydXJIVbb3ISdHFQ27wlq3b0kT9huLe0WW17HGv8soxgTT3iJ8Bb7q5PTzLwxy7ZsLKZtPvctimAZ0Kf1tgLEwMiXeF1rzaeghzhbk5tYGSP8hmkpfNmosFtWTNvLZzFreS8ze1pUUUO6pG7ZfBDiSLkHN3nfR6LVpV3egWHofK60a7OkId9e0Ijk8wuU9p7PEpBXQI8DXeZ9WiutPuJA=",
                        "expiration": 1734970968
                    }
                }
            }
        }
    },
]
```
Please, see related discussion in [slack](https://expensify.slack.com/archives/C06ML6X0W9L/p1734956543574499?thread_ts=1734560637.707449&cid=C06ML6X0W9L).

Reply:
Thanks